File: T8. DATA TYPES AND VARIABLES - EXERCISES (Java)/P18_DifferentIntegersSize.java
import java.util.Scanner;

public class P18_DifferentIntegersSize {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);

        String input = in.nextLine();
        String result = "";

        try {
            byte a = Byte.parseByte(input);
            result += "* byte\n";
        } catch (Exception ex) {
        }
        try {
            short b = Short.parseShort(input);
            result += "* short\n";
        } catch (Exception ex) {
        }
        try {
            int c = Integer.parseInt(input);
            result += "* int\n";
        } catch (Exception ex) {
        }
        try {
            long d = Long.parseLong(input);
            result += "* long";
        } catch (Exception ex) {
        }

        if (result.length() != 0) {
            System.out.println(input + " can fit in:");
            System.out.print(result);
        } else {
            System.out.println(input + " can't fit in any type");
        }
    }
}

File: T8. DATA TYPES AND VARIABLES - EXERCISES/P18.DifferentIntegersSize.cs
using System;

namespace P18.DifferentIntegersSize
{
    class Program
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();
            string result = "";

            try
            {
                sbyte a = sbyte.Parse(input);
                result += "* sbyte" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                byte b = byte.Parse(input);
                result += "* byte" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                short c = short.Parse(input);
                result += "* short" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                ushort d = ushort.Parse(input);
                result += "* ushort" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                int e = int.Parse(input);
                result += "* int" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                uint f = uint.Parse(input);
                result += "* uint" + Environment.NewLine;
            }
            catch (Exception)
            {
            }
            try
            {
                long g = long.Parse(input);
                result += "* long";
            }
            catch (Exception)
            {
            }

            if (result.Length != 0)
            {
                Console.WriteLine($"{input} can fit in:");
                Console.WriteLine(result);
            }
            else
            {
                Console.WriteLine($"{input} can't fit in any type");
            }
        }
    }
}
